open_flamingo/train/train_utils.py

```diff
@@ -375,3 +376,24 @@ def save_checkpoint(model, optimizer, lr_scheduler, epoch, args):
         if args.delete_previous_checkpoint:
             if epoch > 0:
                 os.remove(f"{args.run_name}/checkpoint_{epoch-1}.pt")
+
+
+def save_deepspeed_checkpoint(model, epoch, args):
```

Comment:
TODO for me: Fix deepspeed model saving when using gradient checkpointing. Currently the issue is that when exclude_frozen_parameters is true the code tries to remove parameters but runs into an inconsistency with naming as grad checkpointed parameters are wrapped.